Userland/Libraries/LibWeb/DOM/Document.h

```diff
@@ -237,13 +237,15 @@ class Document
     const String& content_type() const { return m_content_type; }
     void set_content_type(const String& content_type) { m_content_type = content_type; }
 
-    const String& encoding() const { return m_encoding; }
+    bool has_encoding() const { return m_encoding.has_value(); }
+    const Optional<String>& encoding() const { return m_encoding; }
+    String encoding_or_default() const { return m_encoding.value_or("UTF-8"); }
     void set_encoding(const String& encoding) { m_encoding = encoding; }
 
     // NOTE: These are intended for the JS bindings
-    const String& character_set() const { return encoding(); }
-    const String& charset() const { return encoding(); }
-    const String& input_encoding() const { return encoding(); }
+    String character_set() const { return encoding_or_default(); }
+    String charset() const { return encoding_or_default(); }
+    String input_encoding() const { return encoding_or_default(); }
```

Comment:
What are these "JS bindings"? I cannot seem to find any reference to these functions anywhere in the repository, so maybe they could even be removed?

Reply:
Things that are exposed to JS via IDL - you'll find them used in the generated bindings code. Don't remove these, it'll break the build :)